sql/expression/function/polygon.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2020-2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// Polygon is a function that returns a polygon type containing values Y and Y.
+type Polygon struct {
+	args []sql.Expression
+}
+
+var _ sql.FunctionExpression = (*Polygon)(nil)
+
+// NewPolygon creates a new polygon expression.
+func NewPolygon(args ...sql.Expression) (sql.Expression, error) {
+	if len(args) < 1 {
+		return nil, sql.ErrInvalidArgumentNumber.New("Polygon", "1 or more", len(args))
+	}
+	return &Polygon{args}, nil
+}
+
+// FunctionName implements sql.FunctionExpression
+func (l *Polygon) FunctionName() string {
+	return "polygon"
+}
+
+// Description implements sql.FunctionExpression
+func (l *Polygon) Description() string {
+	return "returns a new polygon."
+}
+
+// Children implements the sql.Expression interface.
+func (l *Polygon) Children() []sql.Expression {
+	return l.args
+}
+
+// Resolved implements the sql.Expression interface.
+func (l *Polygon) Resolved() bool {
```

Comment:
For this and others, you can embed `NaryExpression` to get `Resolved` and `Children` for free

Reply:
Also `IsNullable`